metrics/core/types.go

```diff
@@ -119,6 +118,8 @@ func (this *LabeledMetric) GetValue() interface{} {
 }
 
 type MetricSet struct {
+	CreateTime     time.Time
+	ScrapeTime     time.Time
```

Comment:
What about metrics that could be scraped at different times? Volume / filesystem stats will be updated on a different schedule from other stats, as will UserDefinedMetrics.